src/bp/bpfs_recovery.ts

```diff
@@ -0,0 +1,85 @@
+import 'bluebird-global'
+// tslint:disable-next-line:ordered-imports
+import './sdk/rewire'
+// tslint:disable-next-line:ordered-imports
+import './common/polyfills'
+
+import chalk from 'chalk'
+import { FatalError } from 'errors'
+import fs from 'fs'
+import { mkdirpSync } from 'fs-extra'
+import path from 'path'
+import 'reflect-metadata'
+import { container } from './core/app.inversify'
+import Database from './core/database'
+import { GhostService } from './core/services'
+import { TYPES } from './core/types'
+
+export default async (argv, action) => {
+  let ghost: GhostService | undefined
+  let database: Database | undefined
+
+  try {
+    ghost = container.get<GhostService>(TYPES.GhostService)
+    database = container.get<Database>(TYPES.Database)
+
+    const { DATABASE_URL } = process.env
+
+    const useDbDriver = process.env.BPFS_STORAGE === 'database'
+    ghost.initialize(useDbDriver)
+
+    const dbType = DATABASE_URL && DATABASE_URL.toLowerCase().startsWith('postgres') ? 'postgres' : 'sqlite'
```

Comment:
I feel like this piece of code is repeated all over the place, can we move it somewhere and reuse it. If we ever want to add another database option then we'll have to rewrite this all around.

Reply:
Yep sure, i'll move that to the database file directly